OpenSim/Simulation/Manager/Manager.cpp

```diff
@@ -513,6 +498,88 @@ setModel(Model& model)
 // INTEGRATOR
 //-----------------------------------------------------------------------------
 //_____________________________________________________________________________
+/**
+  * Set the integrator.
+  */
+void Manager::setIntegratorMethod(IntegratorMethod integMethod)
+{
+    if (_timeStepper) {
+        std::string msg = "Cannot set a new integrator on this Manager";
+        msg += "after Manager::integrate() has been called at least once.";
+        OPENSIM_THROW(Exception, msg);
+    }
+
+    auto& sys = _model->getMultibodySystem();
+    switch (integMethod) {
+        //case IntegratorMethod::CPodes:
+        //    _integ.reset(new SimTK::CPodesIntegrator(sys));
+        //    break;
+
+        case IntegratorMethod::ExplicitEuler:
+            _integ.reset(new SimTK::ExplicitEulerIntegrator(sys));
+            break;
+
+        case IntegratorMethod::RungeKutta2:
+            _integ.reset(new SimTK::RungeKutta2Integrator(sys));
+            break;
+
+        case IntegratorMethod::RungeKutta3:
+            _integ.reset(new SimTK::RungeKutta3Integrator(sys));
+            break;
+
+        case IntegratorMethod::RungeKuttaFeldberg:
+            _integ.reset(new SimTK::RungeKuttaFeldbergIntegrator(sys));
+            break;
+
+        case IntegratorMethod::RungeKuttaMerson:
+            _integ.reset(new SimTK::RungeKuttaMersonIntegrator(sys));
+            break;
+
+        //case Integrator::SemiExplicitEuler:
+        //    _integ.reset(SimTK::SemiExplicitEulerIntegrator(sys, stepSize));
+        //    break;
+
+        case IntegratorMethod::SemiExplicitEuler2:
+            _integ.reset(new SimTK::SemiExplicitEuler2Integrator(sys));
+            break;
+
+        case IntegratorMethod::Verlet:
+            _integ.reset(new SimTK::VerletIntegrator(sys));
+            break;
+
+        default:
+            std::string msg = "Integrator method not recognized.";
+            OPENSIM_THROW(Exception, msg);
+    }
+}
+
+void Manager::setIntegratorMethod(const std::string& integMethod)
```

Comment:
I know you really want to make it easy for the scripting users to specify the integrator, but having two ways to do it- using and enum and a string is not healthy code. My strong preference is to only have the enumerator interface. This does require enabling MATLAB users to pass an integer value. For example, see how we handle[ `ForceLocation`](https://simtk.org/api_docs/opensim/api_docs/classOpenSim_1_1C3DFileAdapter.html#ac2eb67cb6bac08281775988cd46fa604)  for the `C3DFileAdapter`. Users have to look up the documentation to know what strings to type, so its not like the string interface is any easier for scripting users.

Reply:
We could add just the `enum` case for now with examples and then monitor how many/few problems people have. Thanks for the link to the `ForceLocation` documentation. I'll use that as a guide.

Reply:
Bah, I suppose we should be more consistent but I really do prefer the string interface to the integer. Yes, in both cases, the writer of the code must look at the documentation, but it would be nice if the reader didn't need to do so.